test/functorch/test_control_flow.py

```diff
@@ -2011,13 +2117,196 @@ def chain_fct_different_dim(inp):
+    # as the cond operation is classified to be non-pointwise
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+            # and (params["device"] == torch.device("cpu") or torch.version.hip)
+        ),
+    )
+    def test_associative_scan_cond(
+        self, combine_mode, compile_mode, reverse, device
+    ):
+        def body(x, y):
+            # TODO: Is the cond operation really non-pointwise?
+            val = cond(torch.sum(y) > 0., lambda y: y + 0., lambda y: 1. - y, (y,))
+            return x * val
+
+        scan_fct = compile_mode_helper(associative_scan, compile_mode)
+
+        inp = torch.randn(3, 10, 1, device=device)
+        
+        result = scan_fct(body, inp, dim=0, reverse=reverse, combine_mode=combine_mode)
+        expected_result = _fake_associative_scan(
+            body,
+            inp,
+            0,
+            reverse=reverse,
+        )
+        
+        self.assertEqual(result, expected_result)
+        
+    # TODO: provide an implementation for all compile modes and re-enable all test
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("compile_mode_scan", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("compile_mode_associative_scan", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("reverse", [False, True])
+    @parametrize("reverse_associative_scan", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping combine_mode=pointwise 
+    # as the cond operation is classified to be non-pointwise
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+            # and (params["device"] == torch.device("cpu") or torch.version.hip)
+        ),
+    )
+    def test_scan_associative_scan(
+        self, combine_mode, compile_mode_scan, compile_mode_associative_scan, reverse, reverse_associative_scan, device
+    ):
+        # scan_fct = compile_mode_helper(scan, compile_mode_scan)
+        scan_fct = scan
+        # associative_scan_fct = compile_mode_helper(associative_scan, compile_mode_associative_scan)
+        associative_scan_fct = associative_scan
+        init = torch.randn(10, 5, device=device)
+        inp = torch.randn(3, 10, 5, device=device)
+        
+        def body(x, y):
+            val = associative_scan_fct(get_scan_combine_fn("add", True), y, 0, reverse=reverse_associative_scan, combine_mode=combine_mode)
+            return x + 0., x + val
+            # return x + y, x + y
+        
+        result = scan_fct(body, init, inp, dim=0, reverse=reverse)
+        expected_result = _fake_scan(
+            body,
+            init,
+            inp,
+            0,
+            reverse=reverse,
+        )
+        
+        self.assertEqual(result[1], expected_result[1])
```

Comment:
what happens for result[0]

Reply:
Good catch, I only checked the result[1] in that version, but in the new commit, all results are checked